library/std/src/sys/pal/unix/net.rs

```diff
@@ -563,6 +563,24 @@ impl Socket {
         setsockopt(self, libc::SOL_SOCKET, option, mark as libc::c_int)
     }
 
+    #[cfg(any(target_os = "linux", target_os = "haiku", target_os = "vxworks"))]
+    pub fn todevice(&self) -> io::Result<&CStr> {
+        let buf: [libc::c_char; libc::IFNAMSIZ] =
+            getsockopt(self, libc::SOL_SOCKET, libc::SO_BINDTODEVICE)?;
+        let s: &[u8] = unsafe { core::slice::from_raw_parts(buf.as_ptr() as *const u8, buf.len()) };
+        let ifrname = CStr::from_bytes_with_nul(s).unwrap();
```

Comment:
Bytes with null will return an Interior null runtime error when device names are shorter then 16 characters. 
for example `eno1` would be represented as `[101, 110, 111, 49, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`.
```suggestion
        let ifrname = CStr::from_bytes_until_nul(s).unwrap();
```